### ISSUE TITLE
Ctrl+Key shortcuts not working (Linux)

### ISSUE THREAD

**Issue by beelze** [Genymobile/scrcpy#323]
Just built from scrcpy-9999.ebuild. Ctrl+Key shortcuts not working, instead it acting like plain key is pressed (i.e. `Ctrl+V` entering **v** letter in input field).

**Comment by rom1v** [comment]
Related to https://github.com/Genymobile/scrcpy/issues/300?

It works for me on Debian :confused:

**Comment by beelze** [comment]
No, it's a PC box. Is there any way to debug keypresses to help resolve an issue?

**Comment by rom1v** [comment]
If you can compile from sources, finding the guilty commit would help (since I cannot reproduce).

If `v1.4` works, there are very few commits to test. Anyway you can `git bisect`.

**Comment by beelze** [comment]
Just prepared to bisecting or whatever, but decided to rebuild before and check again.

Well, there are some news: in fact, `LCtrl-key` generally working fine, but `RCtrl+key` – not. (Perhaps last time I've used only RCtrl). For instance, `RCtrl+v` produces `v` in input field; `RCtrl+s` produces `s` in input field and then opens recents.

Another problem: I've found that pasting stops on first non-latin[?] character. I.e. trying to paste `abcЭЮЯ123` produces only `abc`

Funny problem: Almost all non-modifier keypresses are «doubled». I've tried to press `aABbCC` and got `aaaABbbBcCCC` in input field.

BTW, is there any way to get device clipboard to computer?

**Comment by rom1v** [comment]
> Well, there are some news: in fact, `LCtrl-key` generally working fine, but `RCtrl+key` – not.

That's [not expected](https://github.com/Genymobile/scrcpy/blob/v1.5-fixversion/app/src/input_manager.c#L154). RCtrl works for me.

Are you sure your `RCtrl` key works?

> Another problem: I've found that pasting stops on first non-latin[?] character. I.e. trying to paste `abcЭЮЯ123` produces only `abc`

Yes, that's "[expected](https://github.com/Genymobile/scrcpy/blob/v1.5-fixversion/server/src/main/java/com/genymobile/scrcpy/EventController.java#L109-L111)" (only [ASCII chars](https://blog.rom1v.com/2018/03/introducing-scrcpy/#inject-text) are supported).

> Funny problem: Almost all non-modifier keypresses are «doubled». I've tried to press aABbCC and got aaaABbbBcCCC in input field. Backspace and right/left arrow keys are «doubled» too.

Maybe you can help on this issue: https://github.com/Genymobile/scrcpy/issues/109#issuecomment-427276521. I'm not able to reproduce.

> BTW, is there any way to get device clipboard to computer?

See https://github.com/Genymobile/scrcpy/issues/145.

**Comment by beelze** [comment]
> Are you sure your RCtrl key works?

Absolutely. Everyday I'm using `LCtrl+C` but `RCtrl+Ins`, in particular. Here's `xev` output for pressing `LCtrl+c` and, immediately after, `RCtrl+c`, maybe it would be helpful:

````
KeymapNotify event, serial 26, synthetic NO, window 0x0,
    keys:  4294967200 0   0   0   32  0   64  0   0   0   0   0   0   0   0   0   
           0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   

KeyPress event, serial 28, synthetic YES, window 0x8c00001,
    root 0x2a0, subw 0x0, time 112157148, (89,89), root:(2451,902),
    state 0x14, keycode 54 (keysym 0x63, c), same_screen YES,
    XLookupString gives 1 bytes: (03) ""
    XmbLookupString gives 1 bytes: (03) ""
    XFilterEvent returns: False

KeymapNotify event, serial 28, synthetic NO, window 0x0,
    keys:  4294967200 0   0   0   32  0   0   0   0   0   0   0   0   0   0   0   
           0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   

KeyRelease event, serial 28, synthetic YES, window 0x8c00001,
    root 0x2a0, subw 0x0, time 112157326, (89,89), root:(2451,902),
    state 0x14, keycode 54 (keysym 0x63, c), same_screen YES,
    XLookupString gives 1 bytes: (03) ""
    XFilterEvent returns: False

KeymapNotify event, serial 28, synthetic NO, window 0x0,
    keys:  4294967200 0   0   0   0   0   64  0   0   0   0   0   0   2   0   0   
           0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   

KeyPress event, serial 28, synthetic YES, window 0x8c00001,
    root 0x2a0, subw 0x0, time 112160340, (89,89), root:(2451,902),
    state 0x14, keycode 54 (keysym 0x63, c), same_screen YES,
    XLookupString gives 1 bytes: (03) ""
    XmbLookupString gives 1 bytes: (03) ""
    XFilterEvent returns: False

KeymapNotify event, serial 28, synthetic NO, window 0x0,
    keys:  4294967200 0   0   0   0   0   0   0   0   0   0   0   0   2   0   0   
           0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   

KeyRelease event, serial 28, synthetic YES, window 0x8c00001,
    root 0x2a0, subw 0x0, time 112160496, (89,89), root:(2451,902),
    state 0x14, keycode 54 (keysym 0x63, c), same_screen YES,
    XLookupString gives 1 bytes: (03) ""
    XFilterEvent returns: False
````

> Maybe you can help on this issue: #109 (comment). I'm not able to reproduce.

Thorougly checked, I've found that only alpha chars, arrows, Backspace, Del and Enter are «doubled». Digits and punctuation chars are not. Maybe it will help. Looks like a different problem.

**Comment by beelze** [comment]
> That's not expected. RCtrl works for me.

As I mentioned previously, it isn't «not working at all».

> `RCtrl+s` produces `s` in input field and then __opens recents__.

You may find [this](https://github.com/mpv-player/mpv/blob/b2229803946e81b11a75110d70b38e05d666f051/video/out/vo_sdl.c#L589-L597) interesting and relevant.

I've checked `Ctrl+key` in `openboardview` and `mpv` (both `libsdl2` dependent too), both `L/RCtrl` works fine. (Though I almost not looking into code)

**Comment by spvkgn** [comment]
Same here in Ubuntu 18.04 (MATE Desktop) with v1.10 and prebuilt server. Can't paste clipboard - both `LCtrl-v` and `RCtrl-v` just produces `v` letter. And all other `Ctrl-key` not works as well.
